DOCS-6282 [APM]Update python automatic log correlation instructions (#19692)
DataDog/documentation: content/en/tracing/other_telemetry/connect_logs_and_traces/python.md

### Standard library logging

To correlate your [traces][1] with your logs, follow the steps below:

> Step 1) Activate automatic instrumentation through any of the options below:

Reviewer: Do we prefer users to generally use one of these three options? That could be good to call out to not overwhelm them with these choices. Or we could at least mention why you'd use one option over the others.

Author: I didn't want to get into which option to use because then we have to revisit the discussion for any documentation for any APM features/configurations like writing custom span tags, enabling DBM/APM, etc. The main doc https://docs.datadoghq.com/tracing/trace_collection/?tab=containers covers a few approaches already as you scroll through it.

There are actually more variations to set up the Python tracer, but I listed the three most common ones.

> To correlate your [traces][1] with your logs, follow the steps below:
> 
> Step 1) Activate automatic instrumentation through any of the options below:
> - Option 1) Through [Library Injection][5]:
>     1. Add the environment variable `DD_LOGS_INJECTION=true` to the application deployment/manifest file
>     2. Follow the instructions in [Library Injection][5] to set up tracing
> - Option 2) Through `ddtrace-run`:
>     1. Apply the environment variable `DD_LOGS_INJECTION=true`in the environment where the application is running
>     2. Import **ddtrace** into the application
>     3. Run the application with `ddtrace-run` (ie: `ddtrace-run python appname.py`)
> - Option 3) Through `patch`:
>     1. Import **ddtrace** into the application
>     2. Add `ddtrace.patch(logging=True)` to the start of the application code
> 
> Step 2) Update your log format to include the required attributes from the log record.

Reviewer: ```suggestion
To correlate your [traces][1] with your logs, complete the following steps:

  1. [Activate automatic instrumentation](#step-1---activate-automatic-instrumentation).
  2. [Include required attributes from the log record](#step-2---include-required-attributes).

#### Step 1 - Activate automatic instrumentation

Activate automatic instrumentation using one of the following options:
 
Option 1: [Library Injection][5]:
  1. Set the environment variable `DD_LOGS_INJECTION=true` in the application `deployment/manifest` file.
  2. Follow the instructions in [Library Injection][5] to set up tracing.

Option 2: `ddtrace-run`:
  1. Set the environment variable `DD_LOGS_INJECTION=true` in the environment where the application is running.
  2. Import **ddtrace** into the application.
  3. Run the application with `ddtrace-run` (for example, `ddtrace-run python appname.py`).
     
Option 3: `patch`:
  1. Import **ddtrace** into the application.
  2. Add `ddtrace.patch(logging=True)` to the start of the application code.

#### Step 2 - Include required attributes

Update your log format to include the required attributes from the log record.
```

Just reorganized this a bit. I think this is easier to follow:
![Screenshot 2023-09-25 at 1 27 58 PM](https://github.com/DataDog/documentation/assets/84536271/fe70bdf5-2852-4e61-87dc-f9f35b4abac8)

### Standard library logging

To correlate your [traces][1] with your logs, there are two steps:

Step 1) Activate automatic instrumentation through any of the options below:
- Option 1) [Library Injection][5]

> - Option 2) Import **ddtrace** then run the application with `ddtrace-run` (ie: `ddtrace-run python appname.py`)

Reviewer 2: With option 2 customers should set: `DD_LOGS_INJECTION=True` otherwise `ddtrace.patch_all()` will not enable log correlation.

Reviewer 2: Actually I am wrong.

Reviewer 2: This will change in ddtrace 2.0. For `ddtrace<2.0`, `ddtrace-run` will call `ddtrace.patch(logging=True)`. 

To avoid confusion we should always ask customers to set this environment variable if they would like to enable this feature.

Reviewer 2: Docs: https://ddtrace.readthedocs.io/en/latest/advanced_usage.html#patch-logging

Author: Great catch, I'll add that! Does this also apply to Option 1?

Author: Update: I just realized I did reproduce Option 1 recently and can confirm DD_LOGS_INJECTION=true needs to be applied there too

hello()
```

> More information can be found in https://ddtrace.readthedocs.io/en/stable/advanced_usage.html#logs-injection

Reviewer: ```suggestion
To learn more about logs injection, read the [ddtrace documentation][6].
```

[2]: /logs/log_collection/python/#configure-the-datadog-agent
[3]: /logs/log_configuration/processors/#trace-remapper
[4]: /tracing/troubleshooting/correlated-logs-not-showing-up-in-the-trace-id-panel/?tab=custom

> [5]: /tracing/trace_collection/library_injection_local/

Reviewer: ```suggestion
[5]: /tracing/trace_collection/library_injection_local/
[6]: https://ddtrace.readthedocs.io/en/stable/advanced_usage.html#logs-injection
```
